src/OpenLoco/src/Graphics/SoftwareDrawingContext.cpp

```diff
@@ -1451,16 +1451,252 @@ namespace OpenLoco::Gfx
             return basePoint;
         }
 
-        static void drawStringYOffsets(const RenderTarget& rt, const Ui::Point& loc, AdvancedColour colour, const void* args, const int8_t* yOffsets)
+        static void drawStringYOffsets(const RenderTarget& rt, const Ui::Point& loc, AdvancedColour colour, const char* str, const int8_t* yOffsets)
         {
-            registers regs;
-            regs.edi = X86Pointer(&rt);
-            regs.esi = X86Pointer(args);
-            regs.ebp = X86Pointer(yOffsets);
-            regs.cx = loc.x;
-            regs.dx = loc.y;
-            regs.al = colour.u8();
-            call(0x0045196C, regs);
+            // This function has been somewhat simplified removing unreachable parts
+            if (colour.isFE())
+            {
+                assert(false);
+                return;
+            }
+            if (loc.x >= rt.x + rt.width)
+                return;
+
+            if (loc.x < rt.x - 1280)
```

Comment:
I think it might be better to use rt.width here, this code is a bit fuzzy and I believe the correct way to handle this is to use the width of the text, if I'm not mistaken this just deals with the case for when text is ever so slightly behind x (0).

Reply:
We can deal with this later.